app/helpers/automated_tests_helper.rb

```diff
@@ -404,7 +399,8 @@ def self.scripts_to_run(assignment, call_on)
 
   # Request an automated test. Ask Resque to enqueue a job.
   def self.async_test_request(grouping_id, call_on)
-    if files_available? && has_permission?
+    # when called on collection
+    if files_available? && (call_on == "collection" || has_permission?)

```

Comment:
I don't know that `collection` should be checked here. If we look at how this is being called in submisison_collector, the `@current_user` is still an admin or TA, and so should have permission.